packages/native/types/base.d.ts

```diff
@@ -103,11 +105,17 @@ export interface CreateStyledComponent<
    * @typeparam AdditionalProps  Additional props to add to your styled component
    */
   <AdditionalProps extends {} = {}>(
-    ...styles: ArrayInterpolation<
-      ComponentProps &
-        SpecificComponentProps &
-        AdditionalProps & { theme: Theme },
-      StyleType
+    ...styles: Array<
+      | StyledOptions<
```

Comment:
`StyledOptions` shouldn't be allowed here - to the very best of my knowledge. `StyledOptions` can only be provided before styles are accepted (options are passed to the outer~ call):
```js
styled(tag, styledOptions)(...styles)
```

Reply:
Ok. Reverted.